src/core/Pattern_match.ml

```diff
@@ -87,7 +87,7 @@ type t = {
    * TODO? do we want to consider the same match but with different engine
    * as separate matches? or better make them equal for dedup purpose?
    *)
-  engine_kind : Engine_kind.t; [@equal fun _a _b -> true]
+  engine_kind : Engine_kind.flavor; [@equal fun _a _b -> true]
```

Comment:
should be rename to engine_flavor then.
Honestly I don't see the difference between engine kind and engine flavor, so I would
suggest to find better names to more clearly differentiate them.